weave/trace_server/calls_query_builder.py

```diff
@@ -893,6 +926,118 @@ def process_calls_filter_to_conditions(
     return conditions
 
 
+def should_add_predicate_filters(conditions: list[Condition]) -> bool:
+    """Determines if predicate filters should be added to the query.
+
+    We want to return false if we operate on a start call part and an end
+    call part AND there is an OR operand.
+
+    Most calls in the table are merged, we don't care about them. Some
+    calls are still in parts: start, end. Specific fields:
+
+    Start:
+    - started_at
+    - inputs_dump
+    - attributes_dump
+
+    End:
+    - ended_at
+    - output_dump
+    - attributes_dump
+    """
+    if not conditions:
+        return False
+
+    # Check if we have any OR operations between start and end fields
+    for condition in conditions:
+        if not hasattr(condition, "operand"):
+            continue
+        if not isinstance(condition.operand, tsi_query.OrOperation):
```

Comment:
I actually think you can support OR using the design (you would just OR the parts instead of ANDing) them. Might want to add a comment about this since it is more of a feature addition rather than a logical impossibility.